File: app/src/main/java/me/imzack/app/ender/injector/module/ReminderPresenterModule.java
package me.imzack.app.ender.injector.module;

import me.imzack.app.ender.view.contract.ReminderViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class ReminderPresenterModule {

    private final ReminderViewContract mReminderViewContract;
    private final int mPlanListPosition;

    public ReminderPresenterModule(ReminderViewContract reminderViewContract, int planListPosition) {
        mReminderViewContract = reminderViewContract;
        mPlanListPosition = planListPosition;
    }

    @Provides
    ReminderViewContract provideReminderViewContract() {
        return mReminderViewContract;
    }

    @Provides
    int providePlanListPosition() {
        return mPlanListPosition;
    }
}

File: app/src/main/java/net/zackzhang/app/end/injector/module/ReminderPresenterModule.kt
package net.zackzhang.app.end.injector.module

import dagger.Module
import dagger.Provides
import net.zackzhang.app.end.view.contract.ReminderViewContract

@Module
class ReminderPresenterModule(private val mReminderViewContract: ReminderViewContract, private val mPlanListPosition: Int) {

    @Provides
    fun provideReminderViewContract() = mReminderViewContract

    @Provides
    fun providePlanListPosition() = mPlanListPosition
}
